src/encoded/static/components/filegallery.js

```diff
@@ -1134,33 +1134,83 @@ VisualizationControls.defaultProps = {
 // Displays the file filtering controls for the file association graph and file tables.
 
 class FilterControls extends React.Component {
-    constructor() {
-        super();
+    constructor(props, reactContext) {
+        super(props);
+
+        this.navigate = reactContext.navigate;
 
         // Bind this to non-React methods.
         this.handleAssemblyAnnotationChange = this.handleAssemblyAnnotationChange.bind(this);
+        this.setDownloadModalVisibility = this.setDownloadModalVisibility.bind(this);
+        this.handleDownloadClick = this.handleDownloadClick.bind(this);
+
+        //state
+        this.state = { isBatchDownloadOpen: false };
     }
 
     handleAssemblyAnnotationChange(e) {
         this.props.handleAssemblyAnnotationChange(e.target.value);
     }
 
+    setDownloadModalVisibility(isOpen) {
+        this.setState({ isBatchDownloadOpen: isOpen });
+    }
+
+    handleDownloadClick() {
+        const { context, filters } = this.props;
+        const accession = context.accession;
+        const assemblies = filters.assembly && filters.assembly.length > 0 && filters.assembly[0] !== 'All assemblies' ?
+            `&files.assembly=${filters.assembly[0].replace(/\s+/g, '')}` :
```

Comment:
I think it’s better to just always use `encodedURIComponent` for the right-hand-side of all query-string elements, which takes care of all these issues in a standard way. This applies to `output_type`, `file_type`, etc.

Reply:
I used "encodedURIComponent" to make the links more secure. But I left the regex. I do need to remove white space; encodedURIComponent would just URI encode it which was not needed.